frontend/src/metabase/lib/schema_metadata.js

```diff
@@ -791,9 +791,10 @@ export function getIconForField(field) {
   return ICON_MAPPING[getFieldType(field)] || "unknown";
 }
 
-export function getSemanticTypeIcon(semanticType) {
+export function getSemanticTypeIcon(semanticType, fallback = "unknown") {
   const semanticTypeMetadata = field_semantic_types_map[semanticType];
-  return semanticTypeMetadata?.icon;
+  const { icon } = semanticTypeMetadata || {};
+  return icon && icon !== "unknown" ? semanticTypeMetadata.icon : fallback;
```

Comment:
@daltojohnso I've switched this branch to use `getSemanticTypeIcon` 🎉  Thank you! Also added a `fallback` arg